AEP/aep-00004.md

```diff
@@ -0,0 +1,163 @@
+- The shards, which are the instances that contain the data (see [Shard documentation](https://www.mongodb.com/docs/manual/core/sharded-cluster-shards/))
+- The routers (also called mongos), which are the instances that route database operations to the good shard(s). They serve as interface between database clients and the shards. (see [Routers (mongos) documentation](https://www.mongodb.com/docs/manual/core/sharded-cluster-query-router/))
+- The config servers, which store the sharded cluster's metadata : how data is distributed, as well authentication information and RBAC settings. They also manage distributed locks. They are used by the routers to determine how to route queries. 
+
+All these components can be deployed through a MongoDB replicaset, thanks to the replication mechanisms provided by MongoDB. 
+
+### Inside Kubernetes
+In our case, since the MongoDB cluster is deployed within a Kubernetes cluster thanks to a Helm chart, we need to describe the infrastructure Helm creates to manage the different components of the MongoDB cluster.
+
+Bitnami's mongodb-sharded Helm chart offers an out-of-the box infrastructure setup needed to host a sharded MongoDB instance and mainly creates the following Kubernetes objects :
+
+- For the shards, each of them is deployed through a StatefulSet. Each StatefulSet contains the replicas associated with each shard. For example, if you want to deploy 3 shards, each being replicated once, you would have in Kubernetes, 3 StatefulSet's : `shard-0, shard-1, shard-2` each of them containing 2 pods, `data-0` and `data-1`.
+- The config servers are also deployed through a unique StatefulSet managing all the config servers.
+- The routers are deployed through a unique Deployment, since they are stateless proccesses, managing all the mongos instances.
+
+Finally, two Services are created :
+- A headless Service, which is used for the internal functionning of the sharded cluster in Kubernetes. For example, to perform a write operation, the routers need to know how to access to a particular shard pod. (for more information on headless Services, see the [Kubernetes documentation](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services))
+- A "classical" Service, which is used by MongoDB clients to access a mongos instance.
+
+### Back to MongoDB
+Once these components are deployed inside a Kubernetes Cluster, a database must be created and sharding enabled on it. To perform that, a connection to a router instance must be established :
+
+Assuming the Service to access the routers is called "mongodb-sharded" and listening on MongoDB default port:
+```bash
+mongosh -u $MONGODB_ADMIN_USER -p $MONGODB_ADMIN_PASSWORD mongodb://mongodb-sharded:27017
+```
+Then to create a database and enable sharding on it :
+```javascript
+sh.enableSharding("our_database")
```

Comment:
where do you need to use this command?

Reply:
This the standard command used in mongosh. It might be the responsibility of ArmoniK users to enable sharding on their database. @lemaitre-aneo  do you agree ?